src/utils/data_buffer.cpp

```diff
@@ -1,11 +1,25 @@
 #include "data_buffer.h"
 
+#include <iostream>
+#include <new>
+
 DataBuffer::DataBuffer (int num_samples, size_t buffer_size)
 {
     this->buffer_size = buffer_size;
     this->num_samples = num_samples;
-    data = new double[buffer_size * num_samples];
     first_free = first_used = count = 0;
+
+    try
+    {
+        data = new double[buffer_size * num_samples];
+    }
+    catch (const std::bad_alloc &e)
+    {
+        std::cerr << "Not enough memory to allocate data buffer with " << num_samples
```

Comment:
lets remove this print

Reply:
To prevent a compiler warning about unused variables, I'm going to cast the exception to void (a common idiom - or so the internet tells me). I could pass the exception by value, but then LGTM will raise a flag on it. Happy to adjust if you have a preferred idiom for marking a variable as unused.

Reply:
https://github.com/brainflow-dev/brainflow/blob/master/third_party/SimpleBLE/src/PeripheralSafe.cpp#L20 you can do like here, just dont declare the variable